### PR TITLE
Wait for given image during cluster upgrades - master

### PR DESCRIPTION
**Pull request by mgencur**
* the condition is different when specifying a concrete image vs. OCP
channel

### CODE REVIEW

File: test/lib.bash [hunk 1]
```diff
@@ -325,21 +324,20 @@ function end_prober {
...
 function upgrade_ocp_cluster {
-  local ocp_target_version upgrade_ocp_image latest_cluster_version
+  local upgrade_ocp_image latest_cluster_version
   upgrade_ocp_image="${1:-}"
 
   if [[ -n "$upgrade_ocp_image" ]]; then
-    ocp_target_version="$upgrade_ocp_image"
     oc adm upgrade --to-image="${UPGRADE_OCP_IMAGE}" \
       --force=true --allow-explicit-upgrade || return $?
+    timeout 7200 "[[ \$(oc get clusterversion version -o jsonpath='{.status.history[?(@.image==\"${upgrade_ocp_image}\")].state}') != Completed ]]" || return $?
```
Comment (skonto) [inline test/lib.bash:333]: is this 2h time? it takes that long?

Comment (mgencur) [reply test/lib.bash:333]: It takes something between half an hour and two hours. The time is always different. Anyway, cluster upgrades don't run in CI. We only re-use this to run cluster upgrade tests in downstream.

**Review by cardil — APPROVED** [review]
/lgtm

**Pull request merged**